integration/yarn/src/main/java/tachyon/yarn/Client.java

```diff
@@ -273,10 +272,10 @@ private void checkClusterResource(GetNewApplicationResponse appResponse) {
   // Checks that there are enough nodes in the cluster to run the desired number of workers
   private void checkNodesAvailable() throws YarnException, IOException {
     Set<String> hosts = YarnUtils.getNodeHosts(mYarnClient);
-    Preconditions.checkArgument(mNumWorkers <= hosts.size(),
-        "Number of workers specified above number of usable hosts in the cluster, "
-            + String.format("specified=%s, but there are only %d usable hosts: %s", mNumWorkers,
-                hosts.size(), hosts));
+    Preconditions.checkArgument(mNumWorkers <= hosts.size() * mMaxWorkersPerHost,
```

Comment:
We don't currently have unit tests for `tachyon.yarn.Client`, so I've created this jira: https://tachyon.atlassian.net/browse/TACHYON-1503